### DISCUSSION TITLE
Two levels backend navigation on custom main menus

### DISCUSSION THREAD

**Discussion by hirisov** [wintercms/winter#196] [Q&A] [answered]
Hello,

we would like to use the two levels (accordion) hierarchy for backend navigation in our custom backend menus, just like it is in the Settings menu. At https://wintercms.com/docs/plugin/registration#navigation-menus we see how to register a custom main menu item ("blog"), and  and the submenus ("posts", "categories"), but this is only one level navigation inside a main menu. 
Considering in Settings it is already implemented to support another level to group the menu items ("cms", "system", "logs" etc) we think it would be very useful feature for bigger projects with many backend lists to support such feature in the custom main menus.

regards, Gabor Horn

**Comment by bennothommo** [comment, marked as answer]
You can use the [NavigationManager::registerContextSidenavPartial()](https://github.com/wintercms/winter/blob/bf91db6e9e74e62c362ebb08ed6abb537b615d5d/modules/backend/classes/NavigationManager.php#L733) method to define a custom partial for the side navigation for a particular main menu item. This is what the [Settings area](https://github.com/wintercms/winter/blob/bf91db6e9e74e62c362ebb08ed6abb537b615d5d/modules/system/ServiceProvider.php#L393) does.

The custom partial can then be set up to format the side menu in whichever fashion you would like. You can take a look at the [Settings sidenav partials](https://github.com/wintercms/winter/tree/develop/modules/system/partials) to get an idea of how to set it up similar to the Settings area.

**Reply by hirisov** [reply]
Thanks a lot for the quick and precise answer!

**Reply by hirisov** [reply]
I also think it would be very useful to add this information somewhere to the https://wintercms.com/docs/plugin/registration#navigation-menus part in the docs, maybe just in such a compact form to give some hint where to look for example if somebody needs a custom logic for their sidenav.

**Reply by bennothommo** [reply]
Good idea, will try and whip something up shortly.
